improver/calibration/rainforest_calibration.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2021 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""RainForests calibration Plugins."""
+
+import warnings
+
+import numpy as np
+from iris.cube import Cube, CubeList
+
+from improver import BasePlugin, PostProcessingPlugin
+
+# Passed to choose_set_of_percentiles to set of evenly spaced percentiles
+DEFAULT_ERROR_PERCENTILES_COUNT = 19
+DEFAULT_OUTPUT_REALIZATIONS_COUNT = 100
+
+
+class TrainRainForestsTreeModels(BasePlugin):
+    """Class to train tree models for use in RainForests calibration."""
+
+
+class ApplyRainForestsCalibration(PostProcessingPlugin):
+    """Class to calibrate input forecast given a series of RainForests tree models."""
+
+    def __init__(self, model_config_dict: dict, nthreads: int):
+        """Initialise the tree model variables used in the application of RainForests
+        Calibration.
+
+        Args:
+            model_config_dict:
+                Dictionary containing Rainforests model configuration variables.
+            nthreads:
```

Comment:
I'll double check this, but as it stands the docstring is inconsistent as you've highlighted. I'll update the docstring, or fix the intended use to align with the above.

Reply:
The intention of this parameter is the number of threads to use for prediction processing, rather than initialisation. So these parameters are being used as intended, but the docstring didn't reflect this. I've updated the docstring for `threads` to:
```
threads:
    Number of threads to use during prediction with tree-model objects.
```